gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/SimpleConfigStoreResolver.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.ConfigStoreWithResolution;
+import gobblin.config.store.api.VersionDoesNotExistException;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValue;
+
+/**
+ * Utility class to provide {@ConfigStoreWithResolution} functionality based on given {@ConfigStore}
+ * @author mitu
+ *
+ */
+public class SimpleConfigStoreResolver implements ConfigStoreWithResolution {
+
+  private static final Logger LOG = Logger.getLogger(SimpleConfigStoreResolver.class);
+  private final ConfigStore store;
+
+  public SimpleConfigStoreResolver(ConfigStore base) {
+    this.store = base;
+  }
+
+  @Override
+  public String getCurrentVersion() {
+    return store.getCurrentVersion();
+  }
+
+  @Override
+  public URI getStoreURI() {
+    return store.getStoreURI();
+  }
+
+  @Override
+  public Collection<URI> getChildren(URI uri, String version) throws VersionDoesNotExistException{
+    return store.getChildren(uri, version);
+  }
+
+  @Override
+  public List<URI> getOwnImports(URI uri, String version) throws VersionDoesNotExistException{
+    return store.getOwnImports(uri, version);
+  }
+
+  @Override
+  public Config getOwnConfig(URI uri, String version) throws VersionDoesNotExistException{
+    return store.getOwnConfig(uri, version);
+  }
+
+  @Override
+  public Config getResolvedConfig(URI uri, String version) throws VersionDoesNotExistException{
+    UUID traceID = UUID.randomUUID();
+    LOG.info(String.format("Getting resolved config for uri: %s for version %s, with traceID: %s", 
+        uri, version, traceID));
+    return getResolvedConfigWithTrace(uri, version, traceID.toString());
+  }
+
+  protected Config getResolvedConfigWithTrace(URI uri, String version, String traceID) throws VersionDoesNotExistException{
+    CircularDependencyChecker.checkCircularDependency(store, version, uri);
+    Config self = this.getOwnConfig(uri, version);
+    for (Entry<String, ConfigValue> entry : self.entrySet()) {
+      LOG.info(String.format("key %s, value %s from uri %s, with traceID %s", 
```

Comment:
Instead of using `String.format` you can do `LOG.info("key %s, value %s from uri %s, with traceID %s", entry.getKey(), ...)`. It is faster, as it only assembles the string if it is needed for logging.